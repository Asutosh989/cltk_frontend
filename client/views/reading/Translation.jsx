Translation = React.createClass({

	propTypes: {
		translation: React.PropTypes.object.isRequired,
	},

	render() {
		const translationClassName = 'meta-item translation';

		return (
			<div className={translationClassName}>
				{this.props.translation.text.map((text, i) => (
					<p
						key={i} data-num={i} className="translation-text"
						dangerouslySetInnerHTML={{ __html: text }}
					/>
				))}

			</div>

		);
	},

});
