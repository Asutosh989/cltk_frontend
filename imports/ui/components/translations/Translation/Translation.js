import React from 'react';
import PropTypes from 'prop-types';


class Translation extends React.Component {

	render() {
		const translationClassName = 'meta-item translation';

		return (
			<div className={translationClassName}>
				{this.props.translation.text.map((text, i) => (
					<p
						key={i} data-num={i} className="translation-text"
						dangerouslySetInnerHTML={{ __html: text }}
					/>
				))}
			</div>
		);
	}
}

Translation.propTypes = {
	translation: PropTypes.object.isRequired,
};


export default Translation;
