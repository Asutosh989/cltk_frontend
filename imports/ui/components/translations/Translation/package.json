{
  "name": "Translation",
  "version": "0.0.1",
  "private": true,
  "main": "./Translation.js"
}
