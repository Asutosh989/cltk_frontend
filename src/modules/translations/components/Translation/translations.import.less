.translations {
  padding:20px 5% 100px;

  p.translation-text {
    font-size:16px;
    font-family:@font-body;
    font-weight:300;
    line-height:1.5em;
    color:#555;
    padding:5px 0px;
    margin:0 6px;
    word-wrap:break-word;

  }

}


.translations-options {
  position:absolute;
  bottom:0;
  left:0;
  width:100%;
  background:rgba(255,255,255,0.97);
  border-top:1px solid @divider-light;
  height:46px;

  .bottom-gradient {
    bottom:auto;
    top:-20px;
  }

}
