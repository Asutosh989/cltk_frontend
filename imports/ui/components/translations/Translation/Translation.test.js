
import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import Translation from './Translation';

describe('Translation', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<MuiThemeProvider>
					<Translation />
				</MuiThemeProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
