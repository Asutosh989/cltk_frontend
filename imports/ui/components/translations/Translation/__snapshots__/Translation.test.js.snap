// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Translation renders correctly 1`] = `
<div
  className="meta-item translation"
/>
`;
